### DISCUSSION TITLE
fluxV2 source-controller http and https proxy setup

### DISCUSSION THREAD

**Discussion by RaimundasR** [fluxcd/flux2#3722] [Q&A] [answered]
Is it possible to use flux components `source-controller` , `helm-controller` as not GitOps and set to the source-controller to use https and http proxy, is necessary because our cluster works behind corporate proxies and the source-controller is unable to reach the helm chart repositories over https.. I can't find in flux documentation about this anywhere except [Kustomization](https://fluxcd.io/flux/cheatsheets/bootstrap/#using-https-proxy-for-egress-traffic) , which is already part of the GitOps and not in the scope of this paragraph -- [use_flux_hr_without_GitOps](https://fluxcd.io/flux/faq/#can-i-use-flux-helmreleases-without-gitops)

 very appreciate for advice and suggestions!! :pray:

**Comment by makkes** [comment, marked as answer]
You can patch the Flux Deployments to pass in the HTTP_PROXY and HTTPS_PROXY environment variables: https://fluxcd.io/flux/cheatsheets/bootstrap/#using-https-proxy-for-egress-traffic

**Reply by RaimundasR** [reply]
ok solution would be something like this right?
```
kubectl patch deployment source-controller -n flux-system --type='json' -p='[{"op": "add", "path": "/spec/template/spec/containers/0/env", "value":[{"name": "HTTP_PROXY", "value": "<http_proxy_value>"},{"name": "HTTPS_PROXY", "value": "<http_proxy_value>" }]}]'
```

**Reply by makkes** [reply]
I would recommend to properly prepare manifest files but yes, to patch the deployments in-cluster this is what you would do.
